### ISSUE TITLE
Mails are not relayed if there is a local user

### ISSUE THREAD

**Issue by Clevero** [modoboa/modoboa#1616]
# Impacted versions

* Modoboa: 1.12.2
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce
* Create a relay domain mydomain.net
* create a normal user  alice@mydomain.net
* send a mail to alice@mydomain.net and bob@mydomain.net (who has not an user on modoboa)

# Current behavior
* Mail is not relayed to alice@mydomain.net but stored in his local mailbox (according to webmail)
* Mail is relayed to bob@mydomain.net

# Expected behavior
All emails are being relayed for a relay domain, if they are CLEAN, wether their is a user on modoboa or not 

I'm currently setting up modoboa as a mail gateway with spam and malware filtering. All mails should be relayed if they are CLEAN and if not stored in the amavis quarantine. A user itself should be able to release mails out of his own quarantine if needed. So I created an user alice@mydomain.net and tested the filtering process.
Currently we do not need local mailboxes because a Microsoft Exchange Server is our actual mail server.

According to a comment in an issue: 
```So, if you create a user with a mailbox for your domain, it means it won't be routed.```
https://github.com/modoboa/modoboa/issues/1131#issuecomment-305484450
Is there any workaround for this? Can I force relaying mails?

I'm not 100% sure if I understood it correctly, but would the SMTP auth plugin like described in https://github.com/modoboa/modoboa/issues/986 solve my problem?

**Comment by tonioo** [comment]
@Clevero Indeed, modoboa allows the creation of accounts on a relay domain for "splitted domain" situations, for example when you migrate a domain to a new server. SMTP plugin won't solve your issue since it will create local accounts, which leads to the same configuration.
Looks like the proper way to fix this would be to allow the creation of a SimplerUser without a mailbox... Need to think about it a bit more.

**Comment by kytv76** [comment]
> Is there any solution for this by now? I would need this too.
Cheers,
rasibo
